srv/components/controller/files/scripts/provision.sh

```diff
@@ -1747,3 +1747,73 @@ do_provision()
             echo "No provisioning details found on the controller"
     }
 }
+
+ntp_status_get()
+{
+    ## objects name in the xml
+    _xml_obj_bt="ntp-status"
+    # Property value to be extracted from xml
+    _xml_obj_plist=("ntp-status" "ntp-server-address" "ntp-contact-time")
+
+    _tmp_file="$tmpdir/ntp"
+    _ntpinfo="$tmpdir/ntp_info"
+    [ -f $_tmp_file ] && rm -rf $_tmp_file
+    [ -f $_ntpinfo ] && rm -rf $_ntpinfo
+
+    _cmd="show ntp-status"
+    echo "ntp_status_get(): Entry" >> $logfile
+    echo "ntp_status_get(): running command: '$_cmd'" >> $logfile
+    cmd_run "$_cmd" 
+    parse_xml $xml_doc $_xml_obj_bt "${_xml_obj_plist[@]}" > $_tmp_file
```

Comment:
Would be better to parse JSON output.

Reply:
I have not written any parse, the utility already has implementation for the xml parser.